src/main/settings/settings.cpp

```diff
@@ -972,6 +972,24 @@ Value MaximumMemorySetting::GetSetting(ClientContext &context) {
 	return Value(StringUtil::BytesToHumanReadableString(config.options.maximum_memory));
 }
 
+//===--------------------------------------------------------------------===//
+// Maximum Temp Directory Size
+//===--------------------------------------------------------------------===//
+void MaximumTempDirectorySize::SetGlobal(DatabaseInstance *db, DBConfig &config, const Value &input) {
+	// FIXME: should this not use 'SetExplicit' when the value is 0?
+	// So it acts as RESET instead when 0 is passed?
```

Comment:
-1 should act as unlimited
0 as 0

Reply:
I let -1 act as reset, so it uses the default value (90% of disk space if temp directory is set)